templates/digital-asset-template/frontend/pages/Mint/hooks/useMintData.ts

```diff
@@ -0,0 +1,98 @@
+import { aptosClient } from "@/utils/aptosClient";
+import { useQuery } from "@tanstack/react-query";
+
+export interface Token {
+  token_name: string;
+  cdn_asset_uris: {
+    cdn_image_uri: string;
+    asset_uri: string;
+  };
+}
+
+export interface Collection {
+  collection_id: string;
+  collection_name: string;
+  current_supply: number;
+  max_supply: number;
+  uri: string;
+  description: string;
+}
+
+interface MintQueryResult {
+  current_collections_v2: Array<Collection>;
+  current_collection_ownership_v2_view: {
+    owner_address: string;
+  };
+  current_collection_ownership_v2_view_aggregate: {
+    aggregate: {
+      count: number;
+    };
+  };
+  current_token_datas_v2: Array<Token>;
+}
+
+export function useMintData(collection_id: string) {
+  return useQuery({
+    queryKey: ["app-state", collection_id],
+    refetchInterval: 1000 * 30,
+    queryFn: async () => {
+      const res = await aptosClient().queryIndexer<MintQueryResult>({
+        query: {
+          variables: {
+            collection_id,
+          },
+          query: `
```

Comment:
any reason not to use https://github.com/aptos-labs/aptos-ts-sdk/blob/main/src/api/digitalAsset.ts#L96 ?

Reply:
No _good_ reason beyond me being lazy and not reading the docs for the SDK 😅.